### PR TITLE
move country data to its own JSON file (fixes #2)

### LINKED ISSUE #2: move data into it's own json
that way others can use it, too.

### PR DESCRIPTION
**Pull request by cvan**

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
needs a rebas'ing

**Comment by cvan** [comment]
updated

File: my.js [hunk 1]
```diff
@@ -1,116 +1,53 @@
...
+// To add a new country, update `countries.json`. To find its country
+// code: http://www.worldatlas.com/aatlas/ctycodes.htm
+
+var xhr = new XMLHttpRequest();
+xhr.open('get', 'countries.json', true);
+xhr.addEventListener('load', renderMap, false);
+xhr.addEventListener('error', function (e, err) {
+  console.error('An error occurred fetching country data: ' + err);
+}, false);
+xhr.send();
```
Comment (nickdesaulniers) [inline my.js:13]: Some thoughts on this block.
Constructors with empty arg lists don't need parens. :eggplant: 
The third argument to `XMLHttpRequest.prototype.open` is optional, defaults to true, you may omit it.
The third argument to EventTarget.addEventListener is optional, defaults to false, you may omit it.
No need to log on error.
If you set the responseType attribute of the XMLHttpRequest instance to `'json'`, you shouldn't need the `try { JSON.parse } catch` block below.
:eggplant:  :eggplant:  :eggplant:

File: my.js [hunk 2]
```diff
@@ -1,116 +1,53 @@
...
   }
-});
 
+  // Transform each object so it matches the format `Datamap` expects.
+  Object.keys(countries).forEach(function (key) {
+    countries[key] = {
+      fillKey: 'SOLD',
+      date: countries[key].date
+    };
+  });
```
Comment (nickdesaulniers) [inline my.js:24]: Can we use `Array.prototype.map` rather than `Array.prototype.forEach` to signify a transform?

Comment (cvan) [inline my.js:24]: sure

File: my.js (hunk 1 above)
Comment (cvan) [inline my.js:13]: I always thought `async` defaulted to `false`. but you're right; it does default to `true`. and good call on just sending the request header.

all good feedback! thanks!

**Comment by cvan** [comment]
updated

**Comment by cvan** [comment]
the sad consequence of this is one additional HTTP request. I had considered maybe adding a build step (via a Gulp task) to inline the country data in `my.js`. but this will do, :pig:. this will do.

**Comment by nickdesaulniers** [comment]
Ah, sorry my advice about using map was bad.  We're iterating keys of an object, not an array of objects.  My bad.  Can you change back to `forEach`. sorry.  Maybe it would be simpler to have an array of objects with the keys `countryCode` and `date`. :eggplant:

**Comment by nickdesaulniers** [comment]
Also, I think we still want to parse the date into a Date object and call `toLocaleDateString` on it, otherwise most people outside of 'Murica will get confused.

**Comment by cvan** [comment]
> Ah, sorry my advice about using map was bad. We're iterating keys of an object, not an array of objects. My bad. Can you change back to forEach. sorry. Maybe it would be simpler to have an array of objects with the keys countryCode and date. :eggplant:

yeah, maps of objects don't exist. yeah, I was considering having the JSON data being an array. but that's considered [bad practice](http://haacked.com/archive/2009/06/25/json-hijacking.aspx/). I don't think you can actually find an exploit in today's browsers, so meh.

**Comment by cvan** [comment]
updated

**Comment by nickdesaulniers** [comment]
Thanks @cvan !

**Pull request merged**